From 3a23346f04de096aa502eb08319ba3b6368c21ea Mon Sep 17 00:00:00 2001
From: Stuart Menefy <stuart.menefy@codasip.com>
Date: Thu, 29 Feb 2024 14:49:10 +0000
Subject: [PATCH 05/11] riscv: dts: codasip: Add Hobgoblin qemu support

Add support for a variant of the Hobgoblin which can be run on the
Hobgoblin qemu platform, which adds virtio devices.

Signed-off-by: Stuart Menefy <stuart.menefy@codasip.com>
---
 arch/riscv/boot/dts/codasip/Makefile          |  1 +
 .../boot/dts/codasip/a70x-hobgoblin-qemu.dts  | 31 +++++++++++++++++++
 .../riscv/boot/dts/codasip/a70x-hobgoblin.dts | 11 +++++--
 .../configs/codasip-a70x-hobgoblin_defconfig  |  5 +--
 4 files changed, 44 insertions(+), 4 deletions(-)
 create mode 100644 arch/riscv/boot/dts/codasip/a70x-hobgoblin-qemu.dts

diff --git a/arch/riscv/boot/dts/codasip/Makefile b/arch/riscv/boot/dts/codasip/Makefile
index c17a19aa5395..79c1066932c2 100644
--- a/arch/riscv/boot/dts/codasip/Makefile
+++ b/arch/riscv/boot/dts/codasip/Makefile
@@ -1,2 +1,3 @@
 # SPDX-License-Identifier: GPL-2.0
 dtb-$(CONFIG_ARCH_CODASIP) += a70x-hobgoblin.dtb
+dtb-$(CONFIG_ARCH_CODASIP) += a70x-hobgoblin-qemu.dtb
diff --git a/arch/riscv/boot/dts/codasip/a70x-hobgoblin-qemu.dts b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-qemu.dts
new file mode 100644
index 000000000000..531137a302fa
--- /dev/null
+++ b/arch/riscv/boot/dts/codasip/a70x-hobgoblin-qemu.dts
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2024 Codasip Limited
+ *
+ * A730 Hobgoblin FPGA platform emulation in qemu.
+ */
+
+#include "a70x-hobgoblin.dts"
+
+&soc {
+	virtio@70000000 {
+		compatible = "virtio,mmio";
+		reg = <0x70000000 0x200>;
+		interrupts-extended = <&plic 28>;
+	};
+	virtio@70000200 {
+		compatible = "virtio,mmio";
+		reg = <0x70000200 0x200>;
+		interrupts-extended = <&plic 29>;
+	};
+	virtio@70000400 {
+		compatible = "virtio,mmio";
+		reg = <0x70000400 0x200>;
+		interrupts-extended = <&plic 30>;
+	};
+	virtio@70000600 {
+		compatible = "virtio,mmio";
+		reg = <0x70000600 0x200>;
+		interrupts-extended = <&plic 31>;
+	};
+};
diff --git a/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts b/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts
index c6cc574a307c..f26c6e0cde31 100644
--- a/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts
+++ b/arch/riscv/boot/dts/codasip/a70x-hobgoblin.dts
@@ -1,3 +1,10 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (c) 2024 Codasip Limited
+ *
+ * A730 Hobgoblin FPGA platform.
+ */
+
 /dts-v1/;
 
 #include "dt-bindings/gpio/gpio.h"
@@ -58,7 +65,7 @@ soc: soc@0 {
 		#size-cells = <1>;
 		compatible = "simple-bus";
 		/* child-bus-address parent-bus-address*2 length */
-		ranges = <0x0 0x0 0x0 0x70000000>;
+		ranges = <0x0 0x0 0x0 0x80000000>;
 
 		plic: interrupt-controller@40000000 {
 			compatible = "codasip,plic", "riscv,plic0";
@@ -68,7 +75,7 @@ plic: interrupt-controller@40000000 {
 			/* Downstream interrupts - how are they specified and how many are available */
 			interrupt-controller;
 			#interrupt-cells = <1>;
-			riscv,ndev = <7>; /* 1 to 7 inclusive */
+			riscv,ndev = <31>; /* 1 to 31 inclusive */
 
 			/* Upstream interrupts - where are they connected */
 			/* Per core pairs: machine external interrupt, supervisor external interrupt */
diff --git a/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig b/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
index 6e7458992d28..dee405fda3a7 100644
--- a/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
+++ b/arch/riscv/configs/codasip-a70x-hobgoblin_defconfig
@@ -51,7 +51,7 @@ CONFIG_DEVTMPFS_MOUNT=y
 CONFIG_BLK_DEV_LOOP=y
 CONFIG_VIRTIO_BLK=y
 CONFIG_NETDEVICES=y
-# CONFIG_ETHERNET is not set
+CONFIG_VIRTIO_NET=y
 # CONFIG_WLAN is not set
 CONFIG_INPUT_MOUSEDEV=y
 CONFIG_VT_HW_CONSOLE_BINDING=y
@@ -66,6 +66,7 @@ CONFIG_HW_RANDOM=y
 CONFIG_HW_RANDOM_VIRTIO=y
 CONFIG_SPI=y
 CONFIG_SPI_XILINX=y
+# CONFIG_PTP_1588_CLOCK is not set
 CONFIG_PINCTRL=y
 CONFIG_GPIOLIB=y
 CONFIG_GPIO_XILINX=y
@@ -83,7 +84,7 @@ CONFIG_RTC_CLASS=y
 # CONFIG_RTC_NVMEM is not set
 CONFIG_RTC_DRV_GOLDFISH=y
 CONFIG_SYNC_FILE=y
-# CONFIG_VIRTIO_MENU is not set
+CONFIG_VIRTIO_MMIO=y
 # CONFIG_VHOST_MENU is not set
 # CONFIG_IOMMU_SUPPORT is not set
 CONFIG_RPMSG_CHAR=y
-- 
2.39.3

